x-pack/solutions/security/plugins/elastic_assistant/server/lib/attack_discovery/schedules/register_schedule/updateAlertsWithAttackIds.ts

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient } from '@kbn/core/server';
+import { ALERT_ATTACK_IDS } from '../fields';
+
+export interface UpdateAlertsWithAttackIdsParams {
+  esClient: ElasticsearchClient;
+  alertIdToAttackIdsMap: Record<string, string[]>;
+}
+
+export async function updateAlertsWithAttackIds({
+  esClient,
+  alertIdToAttackIdsMap,
+}: UpdateAlertsWithAttackIdsParams) {
+  await esClient.updateByQuery({
+    index: '.alerts-security.alerts-default',
```

Comment:
the alerts index needs to be formed dynamically and take into account current space ID